src/rgw/radosgw-admin/radosgw-admin.cc

```diff
@@ -9095,6 +9118,73 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT::DEDUP_STATS) {
+    using namespace rgw::dedup;
+    rgw::sal::RadosStore *store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+    if (!store) {
+      cerr << "ERROR: this command can only work when the cluster has a RADOS "
+	   << "backing store." << std::endl;
+      return EPERM;
+    }
+    int ret = cluster::collect_all_shard_stats(store, formatter.get(), dpp());
+    if (ret == 0) {
+      formatter->flush(cout);
+    }
+    return ret;
+  }
+
+  if (opt_cmd == OPT::DEDUP_ABORT || opt_cmd == OPT::DEDUP_PAUSE || opt_cmd == OPT::DEDUP_RESUME) {
+    using namespace rgw::dedup;
+    urgent_msg_t urgent_msg;
+    if (opt_cmd == OPT::DEDUP_ABORT) {
+      urgent_msg = URGENT_MSG_ABORT;
+    }
+    else if (opt_cmd == OPT::DEDUP_PAUSE) {
+      urgent_msg = URGENT_MSG_PASUE;
+    }
+    else {
+      urgent_msg = URGENT_MSG_RESUME;
+    }
+
+    rgw::sal::RadosStore *store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+    if (!store) {
+      cerr << "ERROR: this command can only work when the cluster has a RADOS "
+	   << "backing store." << std::endl;
+      return EPERM;
+    }
+    return cluster::dedup_control(store, dpp(), urgent_msg);
+  }
+
+  if (opt_cmd == OPT::DEDUP_RESTART_DRY || opt_cmd == OPT::DEDUP_RESTART) {
+    using namespace rgw::dedup;
+    rgw::sal::RadosStore *store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+    if (!store) {
+      cerr << "ERROR: this command can only work when the cluster has a RADOS "
+	   << "backing store." << std::endl;
+      return EPERM;
+    }
+    dedup_req_type_t dedup_type = dedup_req_type_t::DEDUP_TYPE_NONE;
+    if (opt_cmd == OPT::DEDUP_RESTART_DRY) {
+      dedup_type = dedup_req_type_t::DEDUP_TYPE_DRY_RUN;
+    }
+    else if (opt_cmd == OPT::DEDUP_RESTART) {
+      dedup_type = dedup_req_type_t::DEDUP_TYPE_FULL;
```

Comment:
shouldn't we return `EPRM` in this case indicating that non-try-run are not supported?

Reply:
I am still testing with active dedup, should be set later